tools/generate-wire.py

```diff
@@ -81,40 +81,48 @@ def _typesize(typename):
     1: FieldType('u8')
 }
 
+# It would be nicer if we had put '*1' in spec and disallowed bare lenvar.
+# In practice we only recognize raw lenvar when it's the previous field.
+
+# size := baresize | arraysize
+# baresize := simplesize | lenvar
+# simplesize := number | type
+# arraysize := lenvar '*' simplesize
 class Field(object):
-    def __init__(self, message, name, size, comments, typename=None):
+    def __init__(self, message, name, size, comments, prevname):
         self.message = message
         self.comments = comments
         self.name = name.replace('-', '_')
         self.is_len_var = False
         self.lenvar = None
+        self.num_elems = 1
+
+        # If it's an arraysize, swallow prefix.
+        if '*' in size:
+            self.lenvar = size.split('*')[0].replace('-','_')
```

Comment:
Minor nit: I usually prefer to split once and then check the length for the case distinction. It also gets rid of the multiple `.split()` calls. Also since we replace `-` with `_` in all paths we can move that replace up to become `size = size.replace('-', '_')`